core/common/src/main/java/alluxio/resource/ResourcePool.java

```diff
@@ -129,14 +134,15 @@ public T acquire(Integer time, TimeUnit unit) {
         mTakeLock.unlock();
       }
     } catch (InterruptedException e) {
-      throw new RuntimeException(e);
+      throw new IOException(e);
```

Comment:
RuntimeException is progagated to the caller. The rule here is: when "InterruptedException" happens, it *should* be propagated to the caller since the function does not know how to handle it. Your getBlockLock code just skips such exception which is not acceptable.